backend/moonstreamapi/routes/queries.py

```diff
@@ -0,0 +1,384 @@
+"""
+The Moonstream queries HTTP API
+"""
+import logging
+from typing import Any, Dict, List, Optional, Tuple, Union
+from uuid import UUID
+
+import boto3  # type: ignore
+from bugout.data import BugoutResources, BugoutJournalEntryContent, BugoutJournalEntry
+from bugout.exceptions import BugoutResponseException
+from fastapi import APIRouter, Body, Request
+import requests
+from slugify import slugify  # type: ignore
+
+
+from .. import data
+from ..actions import get_query_by_name
+from ..middleware import MoonstreamHTTPException
+from ..settings import (
+    MOONSTREAM_ADMIN_ACCESS_TOKEN,
+    MOONSTREAM_APPLICATION_ID,
+    MOONSTREAM_CRAWLERS_SERVER_URL,
+    MOONSTREAM_CRAWLERS_SERVER_PORT,
+    MOONSTREAM_QUERIES_BUCKET,
+    MOONSTREAM_QUERIES_JOURNAL_ID,
+    BUGOUT_RESOURCE_QUERY_RESOLVER,
+)
+from ..settings import bugout_client as bc
+
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter(
+    prefix="/queries",
+)
+
+
+BUGOUT_RESOURCE_QUERY_RESOLVER = "query_name_resolver"
+
+
+@router.get("/list", tags=["queries"])
+async def get_list_of_queries_handler(request: Request) -> List[Dict[str, Any]]:
+
+    token = request.state.token
+
+    # Check already existed queries
+
+    params = {
+        "type": BUGOUT_RESOURCE_QUERY_RESOLVER,
+    }
+    try:
+        resources: BugoutResources = bc.list_resources(token=token, params=params)
+    except BugoutResponseException as e:
+        raise MoonstreamHTTPException(status_code=e.status_code, detail=e.detail)
+    except Exception as e:
+        logger.error(
+            f"Error listing subscriptions for user ({request.user.id}) with token ({request.state.token}), error: {str(e)}"
+        )
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    users_queries: List[Dict[str, Any]] = [
+        resource.resource_data for resource in resources.resources
+    ]
+    return users_queries
+
+
+@router.post("/", tags=["queries"])
+async def create_query_handler(
+    request: Request, query_name: str, query_applied: data.PreapprovedQuery = Body(...)
+) -> BugoutJournalEntry:
+    """
+    Create query in bugout journal
+    """
+
+    token = request.state.token
+
+    user = request.state.user
+
+    # Check already existed queries
+
+    params = {
+        "type": BUGOUT_RESOURCE_QUERY_RESOLVER,
+    }
+    try:
+        resources: BugoutResources = bc.list_resources(token=token, params=params)
+    except BugoutResponseException as e:
+        raise MoonstreamHTTPException(status_code=e.status_code, detail=e.detail)
+    except Exception as e:
+        logger.error(
+            f"Error listing subscriptions for user ({request.user.id}) with token ({request.state.token}), error: {str(e)}"
+        )
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    used_queries: List[str] = [

```

Comment:
I think better to wrap this parts of code with `resource.resource_data["name"]` and `slugify(query_applied.name)` in try..catch, or event create function in actions.py